src/torchjd/autogram/diagonal_sparse_tensor.py

```diff
@@ -289,14 +289,45 @@ def unsqueeze_default(t: DiagonalSparseTensor, dim: int) -> Tensor:
 
 @implements(aten.view.default)
 def view_default(t: DiagonalSparseTensor, shape: list[int]) -> Tensor:
+    # TODO: add error message when error is raised
+    # TODO: handle case where the contiguous_data has to be reshaped
+
     assert isinstance(t, DiagonalSparseTensor)
 
-    if shape == list(t.shape):
-        return diagonal_sparse_tensor(t.contiguous_data, t.v_to_p)
-    else:
-        raise ValueError(
-            f"Non-trivial view not supported yet.\n{t.debug_info()}\ntarget shape: {shape}"
-        )
+    if prod(shape) != t.numel():
+        raise ValueError(f"shape '{shape}' is invalid for input of size {t.numel()}")
+
+    new_v_to_p = []
+    idx = 0
+    flat_v_to_p = [dim for dims in t.v_to_p for dim in dims]
+    for s in shape:
+        # Always add the first element of the group, before even entering the while.
+        # This is because both s and t.contiguous_data.shape[flat_v_to_p[idx]] could be equal to 1,
+        # in which case the while will not even be entered but we still want to add the dimension to
+        # the group. More generally, it's a bit arbitrary in which groups the dimension of length 1
+        # are put, but it should rarely be an issue.
```

Comment:
We could add an assert at construction of tensors so as to enforce where the dimensions `1` can be. This should be thought of jointly with `unsqueeze`, `reshape`, `view` and `expend`. It feels like an element of `v_to_p` whose corresponding dim-sizes are `[1, 2, 3]` should not exist and should be `[2,3]` but I might be wrong. This would possibly remove the need to add the first element here.